### ISSUE TITLE
HOCON serialization library not compatible with Java 8

### ISSUE THREAD

**Issue by sczerwinski** [Kotlin/kotlinx.serialization#1189]
**Describe the bug**
It's not possible to use the library in Android projects.
Build fails, because the library requires Java 11:
```
> Could not resolve all files for configuration ':myproject:compileClasspath'.
   > Could not resolve org.jetbrains.kotlinx:kotlinx-serialization-hocon:1.0.1.
     Required by:
         project :myproject
      > No matching variant of org.jetbrains.kotlinx:kotlinx-serialization-hocon:1.0.1 was found. The consumer was configured to find an API of a library compatible with Java 8, preferably in the form of class files, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' but:
          - Variant 'apiElements' capability org.jetbrains.kotlinx:kotlinx-serialization-hocon:1.0.1 declares an API of a library, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm':
              - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
          - Variant 'runtimeElements' capability org.jetbrains.kotlinx:kotlinx-serialization-hocon:1.0.1 declares a runtime of a library, packaged as a jar, and its dependencies declared externally, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm':
              - Incompatible because this component declares a component compatible with Java 11 and the consumer needed a component compatible with Java 8
```

**To Reproduce**
```groovy
apply plugin: 'kotlin'
apply plugin: 'kotlinx-serialization'

compileKotlin {
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_1_8

    kotlinOptions {
        jvmTarget = JavaVersion.VERSION_1_8.toString()
    }
}

dependencies {
    implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.4.10"
    implementation 'org.jetbrains.kotlinx:kotlinx-serialization-hocon:1.0.1'
}
```

**Expected behavior**
Since `com.typesafe:config` is compatible with Java 8, the same should be true for `org.jetbrains.kotlinx:kotlinx-serialization-hocon`. The build should pass.

**Environment**
 - Kotlin version: 1.4.10
 - Library version: 1.0.1
 - Kotlin platforms: JVM
 - Gradle version: 6.7
 - IDE version (if bug is related to the IDE): Android Studio 4.1 (but also reproducible from command line)
 - Other relevant context: Mac OS X 10.15.7 x86_64, JVM 1.8.0_121 (Oracle Corporation 25.121-b13)

**Comment by gildor** [comment]
If somebody still wants to use hocon serialization on Java < 11, you can override metadata:

```
dependencies {
   implementation("org.jetbrains.kotlinx:kotlinx-serialization-hocon:1.0.1")
   components {
        withModule("org.jetbrains.kotlinx:kotlinx-serialization-hocon", TargetJvmVersionRule::class.java) {}
   }
}

open class TargetJvmVersionRule @Inject constructor(): ComponentMetadataRule {
    @Inject
    open fun getObjects(): ObjectFactory? {
        return null
    }

    override fun execute(context: ComponentMetadataContext) {
        context.details.withVariant("apiElements") {
            attributes {
                attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, 8)
            }
        }
        context.details.withVariant("runtimeElements") {
            attributes {
                attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, 8)
            }
        }
    }
}

```

**Comment by osipxd** [comment]
Slightly more compact workaround based on @gildor's answer and [Gradle Docs](https://docs.gradle.org/current/userguide/component_metadata_rules.html):
```kotlin
dependencies {
    components {
        withModule<HoconSerializationRule>("org.jetbrains.kotlinx:kotlinx-serialization-hocon")
    }
    implementation("org.jetbrains.kotlinx:kotlinx-serialization-hocon:[version]")
}

@CacheableRule
abstract class HoconSerializationRule : ComponentMetadataRule {
    override fun execute(context: ComponentMetadataContext) {
        context.details.allVariants {
            attributes {
                attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, 8)
            }
        }
    }
}
```

Or, if you don't want to create new class, you can use this:
```kotlin
dependencies {
    components {
        withModule("org.jetbrains.kotlinx:kotlinx-serialization-hocon") {
            allVariants {
                attributes {
                    attribute(TargetJvmVersion.TARGET_JVM_VERSION_ATTRIBUTE, 8)
                }
            }
        }
    }
    implementation("org.jetbrains.kotlinx:kotlinx-serialization-hocon:[version]")
}
```

**Comment by sandwwraith** [comment]
It should be fixed in 1.2.x, if not, please reopen

**Comment by sandwwraith** [comment]
Ah, judging by #1621 , it's still not. Looks like it somehow automatically inserts Java version that was used to build release jar
